include/Gaffer/Loop.inl

```diff
@@ -379,8 +392,15 @@ const ValuePlug *Loop<BaseType>::sourcePlug( const ValuePlug *output, const Cont
 	}
 	else if( ancestor == outPlugInternal() )
 	{
-		const int iterations = iterationsPlug()->getValue();
-		if( iterations > 0 && enabledPlug()->getValue() )
+		int iterations;
+		bool enabled;
+		{
+			typename LoopTraits<BaseType>::IterationsScope iterationsScope( context );
```

Comment:
Hmm, I thought last I checked, your philosophy was to avoid creating temp contexts except where they actually seem useful?  All the places I've been adding them are on things that it seems very likely you would actually want to drive with an expression.  I can't picture any good reason why you would drive the name of the loop index variable with an expression?  I can add an extra scope if you want me to though.